bpfd/src/server/pull_bytecode.rs

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: (MIT OR Apache-2.0)
+// Copyright Authors of bpfd
+use flate2::read::GzDecoder;
+use log::debug;
+use oci_distribution::{client, manifest, secrets::RegistryAuth, Client, Reference};
+use serde::Deserialize;
+use serde_json::Value;
+use tar::Archive;
+
+const CONTAINERIZED_BYTECODE_PATH: &str = "/etc/bpfd/bytecode/";
+
+#[derive(Debug, Deserialize, Default)]
+pub struct ContainerImageMetadata {
+    #[serde(rename(deserialize = "io.ebpf.program_name"))]
+    pub name: String,
+    #[serde(rename(deserialize = "io.ebpf.section_name"))]
+    pub section_name: String,
+    #[serde(rename(deserialize = "io.ebpf.program_type"))]
+    pub program_type: String,
+    #[serde(rename(deserialize = "io.ebpf.filename"))]
+    pub filename: String,
+}
+
+#[derive(Debug, Deserialize, Default)]
+pub struct ProgramOverrides {
+    pub path: String,
+    pub image_meta: ContainerImageMetadata,
+}
+
+pub async fn pull_bytecode(image_url: &String) -> Result<ProgramOverrides, anyhow::Error> {
+    debug! {"Pulling bytecode from image path: {}", image_url}
+    let image: Reference = image_url
+        .parse()
+        .expect("Not a valid bytecode image reference");
+
+    let protocol = client::ClientProtocol::Https;
+    // TODO(astoycos): Add option/flag to authenticate against private image repositories
```

Comment:
That should live in `bpfd.toml` we can have something like:

```
[ registry ]

[ registry.docker-io ]
username = foo
password = bar
```

We can then infer the registry from the image path and login if required.
Can be added later so lets add an issue to track it,

Reply:
Sounds good to me I will make an issue to track this